tools/tplg_parser/tplg_parser.c

```diff
@@ -135,24 +137,43 @@ int tplg_load_buffer(int comp_id, int pipeline_id, int size,
 		return -EINVAL;
 	}
 
-	ret = fread(array, sizeof(struct snd_soc_tplg_vendor_array), 1, file);
-	if (ret != 1) {
-		fprintf(stderr, "error: fread fail during load_buffer\n");
-		free(array);
-		return -EINVAL;
-	}
+	/* read vendor tokens */
+	while (parsed_size < size) {
+		read_size = sizeof(struct snd_soc_tplg_vendor_array);
+		ret = fread(array, read_size, 1, file);
+		if (ret != 1) {
+			fprintf(stderr,
+				"error: fread fail during load_buffer\n");
+			free(array);
+			return -EINVAL;
```

Comment:
return -errno; otherwise we change the error that is sent up the stack. Best to check other places too.

Reply:
Yep, I'm addressing the malloc() and fseek() functions now. I noticed that I made a mistake with commits. I need to merge commits 2 and 3. The next  update contains these changes.

Reply:
OK, it's done now for the current PR version!